### ISSUE TITLE
UTF-8 decoding error: Not enough space

### ISSUE THREAD

**Issue by Plep-m** [Miisan-png/Love-Dialogue#8]
if you use any accents in the text it crash with the following UTF-8 decoding error: Not enough space error
```lua
local char = self.displayedText:sub(i, i)
 local charWidth = self.font:getWidth(char)
 ```

**Comment by Miisan-png** [comment]
I am currently working on a massive rework of the library will look into this too. Remaking the entire branching system and improving syntax. 
![image](https://github.com/user-attachments/assets/6583c1a8-830f-4bef-8622-74a16e752e1a)
(i just tried to recreate this) Will have to rework draw rendering.

**Comment by Plep-m** [comment]
It looks great, i did try to find my way around the issue I mentioned but didn't find the solution yet. If you are doing a massive rework I'll simply wait until your first pushes, if you need help don't hesitate to ping me on the issues

**Comment by Miisan-png** [comment]
Yeah, Sounds Good, i will push the new changes soon after i am done fixing bugs. Reworking quite a lot of things. And improving the pluging (with auto complete and syntax highlighting) Thanks for the help:))

**Comment by Nrosa01** [comment]
I'm facing the same error. origin-dev branch hasn't been updated in quite a while. I was thinking too in contributing to this library as it will be really important for me, but if you say that you're making a big refactor I will wait too, although I'd love if you could push some dev code so we can see where this is going and maybe we can help troubleshoot and debug it ^^

**Comment by Miisan-png** [comment]
> I'm facing the same error. origin-dev branch hasn't been updated in quite a while. I was thinking too in contributing to this library as it will be really important for me, but if you say that you're making a big refactor I will wait too, although I'd love if you could push some dev code so we can see where this is going and maybe we can help troubleshoot and debug it ^^

hii, i updated main branch , check that. I haven't made any changes after the changes in main branch

**Comment by Miisan-png** [comment]
As for this issue, i have yet to find a solution. I haven't been working on this due to college work. But any idea @Plep-m  where could i look into? Has to do something with love 2d font renderer i believe

**Comment by Nrosa01** [comment]
I don't see any change on the main branch 🤔
On the other hand, I've been reading the whole code of this project, so when I get some time after work I'll try to find and fix the issue.
I've also noticed an issue where text effects like waves behave depending on the screen resolution instead to a relative screen percentage, I don't know if that's intended but given it's easy to walkaround and that you're doing a massive rework I don't consider a new issue necessary.

I'm really interested in this asset as I don't have time to build my own system from scratch, so I'll try to contribute to it in my limited time ><

**Comment by Plep-m** [comment]
> As for this issue, i have yet to find a solution. I haven't been working on this due to college work. But any idea @Plep-m where could i look into? Has to do something with love 2d font renderer i believe

The issue happen when you try rendering of specific characters like "é" , maybe the size of this character is different ? Maybe it's composed of ' + e ? The utf8 thing is still a mystery for me

**Comment by Plep-m** [comment]
It surly have something to do with love2d characters width calculations

**Comment by Nrosa01** [comment]
Yes! UTF-8 strings have variable byte characters, "é" does indeed takes 2 bytes. I still don't understand why this error appears but what is sure is that tu support more characters and avoid this type of errors. There is [utf8](https://www.love2d.org/wiki/utf8) module in love2d that should be used to handle these strings. I will attempt to convert the library to support utf8 strings in an attempt to fix this error and also because I'm spaniard and I need to use these "special" characters

**Comment by Nrosa01** [comment]
I've fixed this problem! Let me review everything to make sure I didn't mess somewhere else but it should be fine as far as I'm concerned. It was as easy as adding utf-8 support in lovedialogue.lua

**Comment by Nrosa01** [comment]
Done in #10! I hope everything there is fine ^^
I'll check now the other issue in this repo now that I have some spare time

**Comment by Plep-m** [comment]
Thanks for the contribution @Nrosa01 we can close this issue now, resolved ! 🎉
